src/App.tsx

```diff
@@ -1,41 +1,81 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import './styles/general.scss';
+import { Post, User } from './react-app-env';
+import { getUserPosts } from './api/post';
+import { getUsers } from './api/users';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [selectedPostId, setSelectedPostId] = useState(0);
+
+  useEffect(() => {
+    getUserPosts(selectedUserId)
+      .then(postsFromServer => setPosts(postsFromServer));
+
+    getUsers()
+      .then(usersFromServer => setUsers(usersFromServer));
+  }, []);
+
+  const setPostId = (postId: number) => {
+    setSelectedPostId(postId);
+  };
+
+  const visiblePosts = posts.filter(post => {
+    if (!selectedUserId) {
+      return post;
+    }
+
+    return post.userId === selectedUserId;
+  });
+
```

Comment:
use correct request to endpoint instead of filter
<img width="1004" alt="image" src="https://user-images.githubusercontent.com/30314756/176456518-1f7ab7ce-4793-4c69-9156-f3e6c981666e.png">